### PR TITLE
support for NotSegment & NotConverter

### PR DESCRIPTION
**Pull request by kanha-gupta**
Ref #24200 .

Changes proposed in this pull request:
  -

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by kanha-gupta** [comment]
@strongduanmu please review :)

File: kernel/sql-federation/optimizer/src/main/java/org/apache/shardingsphere/sqlfederation/optimizer/converter/segment/expression/impl/NotExpressionConverter.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.shardingsphere.sql.parser.sql.common.segment.dml.expr.NotExpression;
+import org.apache.shardingsphere.sqlfederation.optimizer.converter.segment.SQLSegmentConverter;
+import org.apache.shardingsphere.sqlfederation.optimizer.converter.segment.expression.ExpressionConverter;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+
+public final class NotExpressionConverter implements SQLSegmentConverter<NotExpression, SqlNode> {
```
Comment (strongduanmu) [inline kernel/sql-federation/optimizer/src/main/java/org/apache/shardingsphere/sqlfederation/optimizer/converter/segment/expression/impl/NotExpressionConverter.java:32]: Please add javadoc for NotExpressionConverter.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: kernel/sql-federation/optimizer/src/main/java/org/apache/shardingsphere/sqlfederation/optimizer/converter/segment/expression/impl/NotExpressionConverter.java (hunk 1 above)
Comment (kanha-gupta) [reply kernel/sql-federation/optimizer/src/main/java/org/apache/shardingsphere/sqlfederation/optimizer/converter/segment/expression/impl/NotExpressionConverter.java:35]: Done.

**Comment by kanha-gupta** [comment]
@strongduanmu please review again :)

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**